### DISCUSSION TITLE
Choice Domain vs Distinct

### DISCUSSION THREAD

**Discussion by jedi1992** [EvgSkv/logica#145] [Q&A] [answered]
Hi,
Just for the fun of it was implementing some of the examples from Souffle languge in Logica to compare/contrast syntax.  I agree that not all features from Souffle would make sense in Logica.  Do you feell there is a need to have some syntanx in Logica equivalent to Choice domain - https://souffle-lang.github.io/choice

Distinct without aggregate comes close but does not go the full way. Esp would be nice to know how to implement the "Eligible advisors" example in that page ( https://souffle-lang.github.io/choice) in current Logica syntax. 

Choice domain  feels like a random state space search without an objective function but could be an useful operator. Or am I missing something ?

TIA,

**Comment by EvgSkv** [comment, marked as answer]
Hello @jedi1992 !
It looks like this functionality can be achieved in Logica with `AnyValue` aggregator.
Here is a CoLab implementing advisor selection:
https://colab.research.google.com/github/EvgSkv/logica/blob/main/examples/more/Using_AnyValue.ipynb

`AnyValue` aggregator doesn't work in SQLite yet, as there is not corresponding SQLite function. We will need to implement it.

Let me know if I'm missing something in what choice-domain does.
